src/engine/controls/bpmcontrol.cpp

```diff
@@ -944,57 +958,54 @@ void BpmControl::trackLoaded(TrackPointer pNewTrack) {
 void BpmControl::trackBeatsUpdated(mixxx::BeatsPointer pBeats) {
     if (kLogger.traceEnabled()) {
         kLogger.trace() << getGroup() << "BpmControl::trackBeatsUpdated"
-                        << (pBeats ? pBeats->getBpm() : 0.0);
+                        << (pBeats ? pBeats->getBpm() : mixxx::Bpm(0.0));
     }
     m_pBeats = pBeats;
     updateLocalBpm();
     resetSyncAdjustment();
 }
 
 void BpmControl::slotBeatsTranslate(double v) {
-    mixxx::BeatsPointer pBeats = m_pBeats;
-    if (v > 0 && pBeats && (pBeats->getCapabilities() & mixxx::Beats::BEATSCAP_TRANSLATE)) {
-        double currentSample = getSampleOfTrack().current;
-        double closestBeat = pBeats->findClosestBeat(currentSample);
-        int delta = currentSample - closestBeat;
+    if (v > 0 && m_pBeats) {
+        mixxx::Frame currentFrame(getSampleOfTrack().current / 2.0);
+        mixxx::Frame closestBeat = m_pBeats->findClosestBeat(currentFrame);
+        int delta = currentFrame.getValue() - closestBeat.getValue();
```

Comment:
Append the suffix "Samples" or "Frames" to variable names to distinguish them, i.e. `deltaFrames`

Reply:
Since frames are doubles delta should not be an integer!

```
auto deltaFrames = currentFrame.getValue() - closestBeat.getValue();
```